flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java

```diff
@@ -0,0 +1,131 @@
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+public class FlightDataKafkaProducerTests {
+    @Mock
+    private MockProducer<Long, Flight> producer;
+    private FlightDataKafkaProducer flightDataProducer;
+    private String[] validRecord;
+    private String topic = "";
+
+
+    @Before
+    public void setUp() throws IOException {
+        producer = new MockProducer<>(true, new LongSerializer(), null);
+        flightDataProducer = new FlightDataKafkaProducer("", topic, producer);
+        validRecord = new String[]{"test", "test", "Internacional", "departureEstimate", "test", "Realizado",
+                "test", "test", "test", "test", "test", "test", "test", "test", "test",
+                "0.0", "0.0", "0.0", "0.0"};
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void ThrowsExceptionWhenStringArrayIsEmpty() {
+        //given
+        String[] emptyArray = (String[]) Array.newInstance(String.class, 0);
+        // when
+        flightDataProducer.createFlight(emptyArray);
+    }
+
+    @Test
+    public void recordFailsToPassDataValidationWithInvalidFlightType() {
+        //given
+        String[] invalidRecord = validRecord;
+        invalidRecord[2] = "Invalid";
+        //when
+        boolean result = flightDataProducer.isDataValid(invalidRecord);
+        //then
+        assertEquals(result, false);
+    }
+
+    @Test
+    public void recordFailsToPassDataValidationWithInvalidTimeType() {
+        //given
+        String[] invalidRecord = validRecord;
+        invalidRecord[3] = "Invalid";
+        //when
+        boolean result = flightDataProducer.isDataValid(invalidRecord);
+        //then
+        assertEquals(result, false);
+    }
+
+    @Test
+    public void recordFailsToPassDataValidationWithInvalidFlightSituation() {
+        //given
+        String[] invalidRecord = validRecord;
+        invalidRecord[5] = "Invalid";
+        //when
+        boolean result = flightDataProducer.isDataValid(invalidRecord);
+        //then
+        assertEquals(result, false);
+    }
+
+    @Test
+    public void validRecordPassesDataValidation() {
+        //given
+        //when
+        boolean result = flightDataProducer.isDataValid(validRecord);
+        //then
+        assertEquals(result, true);
+    }
+
+    @Test
+    public void flightObjectIsConstructedCorrectly() {
+        //given
+        Flight flight = new Flight("test", "test", FlightType.Internacional, TimeType.departureEstimate, "test",
+                FlightSituation.Realizado, "test", "test", "test", "test", "test", "test", "test", "test", "test",
+                0.0, 0.0, 0.0, 0.0);
+        //when
+        Flight result = flightDataProducer.createFlight(validRecord);
+        //then
+        assertEquals(flight, result);
```

Comment:
Wrong arguments' order

Reply:
Actually that one is in right order :smile: